dao/src/test/java/org/thingsboard/server/dao/service/BaseTenantServiceTest.java

```diff
@@ -275,4 +323,374 @@ public void testSaveTenantWithIsolatedProfileInMonolithSetup() {
         tenant.setTenantProfileId(isolatedTenantProfile.getId());
         tenantService.saveTenant(tenant);
     }
+
+    @Test
+    public void testGettingTenantAddItToCache() {
+        Tenant tenant = new Tenant();
+        tenant.setTitle("My tenant");
+        Tenant savedTenant = tenantService.saveTenant(tenant);
+
+        reset(tenantDao);
+        Objects.requireNonNull(cacheManager.getCache(CacheConstants.TENANTS_CACHE), "Tenant cache manager is null").evict(savedTenant.getId());
```

Comment:
please, get the tenantCache from cacheManager ONCE. Use the @Before setup() method to set a tenantCache